adoc/extensions/sycl_khr_free_function_commands.adoc

```diff
@@ -0,0 +1,879 @@
+
+_Effects (3)_: Equivalent to [code]#q.submit([&](handler& h) { fill(h, ptr,
+pattern, count); })#.
+
+_Effects (4)_: Equivalent to [code]#q.submit([&](handler& h) { fill(h, dest,
+src); })#.
+
+'''
+
+.[apititle]#update_host#
+[source,role=synopsis,id=api:update_host]
+----
+namespace sycl::khr {
+
+template <typename T, int Dims, access_mode Mode>
+void update_host(sycl::handler& h, accessor<T, Dims, Mode, target::device> acc); (1)
+
+template <typename T, int Dims, access_mode Mode>
+void update_host(sycl::queue q, accessor<T, Dims, Mode, target::device> acc);    (2)
+
+}
+----
+_Effects (1)_: Equivalent to [code]#h.update_host(acc)#.
+
+_Effects (2)_: Equivalent to [code]#q.submit([&](handler& h) { update_host(h,
+acc); })#.
+
+'''
+
+.[apititle]#prefetch#
+[source,role=synopsis,id=api:prefetch]
+----
+namespace sycl::khr {
+
+void prefetch(sycl::handler& h, void* ptr, size_t numBytes); (1)
+
+void prefetch(sycl::queue q, void* ptr, size_t numBytes);    (2)
+
+}
+----
+_Effects (1)_: Equivalent to [code]#h.prefetch(ptr, numBytes)#.
+
+_Effects (2)_: Equivalent to [code]#q.submit([&](handler& h) { prefetch(h, ptr,
+numBytes); })#.
+
+'''
+
+.[apititle]#mem_advise#
+[source,role=synopsis,id=api:mem_advise]
+----
+namespace sycl::khr {
+
+void mem_advise(sycl::handler& h, void* ptr, size_t numBytes, int advice); (1)
+
+void mem_advise(sycl::queue q, void* ptr, size_t numBytes, int advice);    (2)
+
+}
+----
+_Effects (1)_: Equivalent to [code]#h.mem_advise(ptr, numBytes, advice)#.
+
+_Effects (2)_: Equivalent to [code]#q.submit([&](handler& h) { mem_advise(h,
+ptr, numBytes, advice); })#.
+
+'''
+
+=== Command and event barriers
+
+.[apititle]#command_barrier#
+[source,role=synopsis,id=api:command_barrier]
+----
+namespace sycl::khr {
+
+void command_barrier(sycl::handler& h); (1)
+
+void command_barrier(sycl::queue q);    (2)
+
+}
+----
+_Effects_: Enqueues a command barrier.
+Any commands submitted after this barrier cannot begin execution until:
+
+* All commands previously submitted to this queue have completed; and
+* All commands associated with this command's dependencies (e.g., via
+  [code]#handler::depends_on#) have completed.
+
+'''
+
+.[apititle]#event_barrier#
+[source,role=synopsis,id=api:event_barrier]
+----
+namespace sycl::khr {
+
+void event_barrier(sycl::handler& h, const std::vector<sycl::event>& events); (1)
+
+void event_barrier(sycl::queue q, const std::vector<sycl::event>& events);    (2)
+
+}
+----
+_Effects_: Enqueues an event barrier.
+Any commands submitted after this barrier cannot begin execution until:
+
+* All commands associated with [code]#events# have completed; and
+* All commands associated with this command's dependencies (e.g., via
+  [code]#handler::depends_on#) have completed.
+
+{note}For both overloads, if [code]#events# is empty and an event barrier has no
+other dependencies (e.g., specified by [code]#handler::depends_on#), then this
+operation is a no-op.{endnote}
+
+'''
+
+== Issues
+
+None.
```

Comment:
Sorry, one more comment.  I propose that we omit the "Issues" section when there are no open issues, especially when we are ready to ratify a KHR.  For example, the "SYCL_KHR_DEFAULT_CONTEXT" specification does not have an "Issues" section.

Reply:
That makes sense.  Removed in fa8a8f6d60e5d3df80c29974b4083a4ea4aa9445.